src/domain/accounts/domain/src/entities/access_token.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+///////////////////////////////////////////////////////////////////////////////
+
+use chrono::{DateTime, Utc};
+use jsonwebtoken::TokenData;
+use opendatafabric::{AccountID, Multihash};
+use serde::{Deserialize, Serialize};
+use uuid::Uuid;
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct KamuAccessToken {
+    pub prefix: String,
+    pub id: Uuid,
+    pub random_bytes: [u8; 16],
+    pub random_bytes_hash: Multihash,
+    pub checksum: u32,
+    pub base32_token: String,
+    pub composed_token: String,
+}
+
+#[derive(Debug, Clone, Eq, PartialEq)]
+pub struct AccessToken {
+    pub id: Uuid,
+    pub token_name: String,
+    pub token_hash: [u8; 32],
+    pub created_at: DateTime<Utc>,
+    pub revoked_at: Option<DateTime<Utc>>,
+    pub account_id: AccountID,
+}
+
+///////////////////////////////////////////////////////////////////////////////
+
+/// Our claims struct, it needs to derive `Serialize` and/or `Deserialize`
+#[derive(Debug, Serialize, Deserialize)]
+pub struct KamuAccessTokenClaims {
+    pub exp: usize,
+    pub iat: usize,
+    pub iss: String,
+    pub sub: String,
+}
```

Comment:
Let's rename this struct to reflect "JWT" type of token and clearly distniguish it from "access token".